In [1]:
# I am using pandas frame to analyse the dataset, so you can see what i meant by solution in steps.
# import pandas dataframe library to read the world_population excel file.
# df is short for dataframe, you can use an variable name you want.

import pandas as pd

df = pd.read_excel('World_Poulation_by_countries.xlsx')
df

Country Name Country Code         2000         2001  \
0                          Aruba          ABW      90866.0      92892.0   
1    Africa Eastern and Southern          AFE  398113044.0  408522129.0   
2                    Afghanistan          AFG   20779957.0   21606992.0   
3     Africa Western and Central          AFW  267214544.0  274433894.0   
4                         Angola          AGO   16395477.0   16945753.0   
..                           ...          ...          ...          ...   
261                       Kosovo          XKX    1700000.0    1701154.0   
262                  Yemen, Rep.          YEM   17409071.0   17918369.0   
263                 South Africa          ZAF   44967713.0   45571272.0   
264                       Zambia          ZMB   10415942.0   10692197.0   
265                     Zimbabwe          ZWE   11881482.0   11923906.0   

            2002         2003         2004         2005         2006  \
0        94992.0      97016.0      98744.0     100028.0     100830.0   
1    419223717.0  430246635.0  441630149.0  453404076.0  465581372.0   
2     22600774.0   23680871.0   24726689.0   25654274.0   26433058.0   
3    281842480.0  289469530.0  297353098.0  305520588.0  313985474.0   
4     17519418.0   18121477.0   18758138.0   19433604.0   20149905.0   
..           ...          ...          ...          ...          ...   
261    1702310.0    1703466.0    1704622.0    1705780.0    1719536.0   
262   18443684.0   18985001.0   19540096.0   20107416.0   20687648.0   
263   46150913.0   46719203.0   47291610.0   47880595.0   48489464.0   
264   10971704.0   11256740.0   11550641.0   11856244.0   12173518.0   
265   11954293.0   11982219.0   12019911.0   12076697.0   12155496.0   

            2007  ...         2012         2013         2014         2015  \
0       101226.0  ...     102565.0     103165.0     103776.0     104339.0   
1    478166911.0  ...  547482863.0  562601578.0  578075373.0  593871847.0   
2     27100542.0  ...   31161378.0   32269592.0   33370804.0   34413603.0   
3    322741656.0  ...  370243017.0  380437896.0  390882979.0  401586651.0   
4     20905360.0  ...   25107925.0   26015786.0   26941773.0   27884380.0   
..           ...  ...          ...          ...          ...          ...   
261    1733404.0  ...    1807106.0    1818117.0    1812771.0    1788196.0   
262   21282514.0  ...   24473176.0   25147112.0   25823488.0   26497881.0   
263   49119766.0  ...   52832659.0   53687125.0   54544184.0   55386369.0   
264   12502958.0  ...   14465148.0   14926551.0   15399793.0   15879370.0   
265   12255920.0  ...   13115149.0   13350378.0   13586710.0   13814642.0   

            2016         2017         2018         2019         2020  \
0       104865.0     105361.0     105846.0     106310.0     106766.0   
1    609978946.0  626392880.0  643090131.0  660046272.0  677243299.0   
2     35383028.0   36296111.0   37171922.0   38041757.0   38928341.0   
3    412551299.0  423769930.0  435229381.0  446911598.0  458803476.0   
4     28842482.0   29816769.0   30809787.0   31825299.0   32866268.0   
..           ...          ...          ...          ...          ...   
261    1777557.0    1791003.0    1797085.0    1788878.0    1790133.0   
262   27168210.0   27834811.0   28498683.0   29161922.0   29825968.0   
263   56207649.0   57009751.0   57792520.0   58558267.0   59308690.0   
264   16363449.0   16853608.0   17351714.0   17861034.0   18383956.0   
265   14030338.0   14236599.0   14438812.0   14645473.0   14862927.0   

            2021  
0       107195.0  
1    694665117.0  
2     39835428.0  
3    470898870.0  
4     33933611.0  
..           ...  
261    1806279.0  
262   30490639.0  
263   60041996.0  
264   18920657.0  
265   15092171.0  

[266 rows x 24 columns]

In [2]:
# I want to describe the dataframe (df) to break down the dataset statistics

df.describe()

2000          2001          2002          2003          2004  \
count  2.650000e+02  2.650000e+02  2.650000e+02  2.650000e+02  2.650000e+02   
mean   2.443804e+08  2.477621e+08  2.511427e+08  2.545388e+08  2.579580e+08   
std    7.753492e+08  7.856437e+08  7.958574e+08  8.060498e+08  8.162574e+08   
min    9.392000e+03  9.478000e+03  9.593000e+03  9.724000e+03  9.871000e+03   
25%    1.228359e+06  1.258008e+06  1.277210e+06  1.283564e+06  1.313492e+06   
50%    8.170172e+06  8.231150e+06  8.427061e+06  8.557160e+06  8.680516e+06   
75%    4.917650e+07  4.866240e+07  4.987167e+07  5.142558e+07  5.306887e+07   
max    6.114324e+09  6.193664e+09  6.272724e+09  6.351856e+09  6.431527e+09   

               2005          2006          2007          2008          2009  \
count  2.650000e+02  2.650000e+02  2.650000e+02  2.650000e+02  2.650000e+02   
mean   2.614031e+08  2.648752e+08  2.683706e+08  2.719208e+08  2.754863e+08   
std    8.265104e+08  8.367913e+08  8.470869e+08  8.575082e+08  8.679735e+08   
min    9.848000e+03  9.827000e+03  9.846000e+03  9.880000e+03  9.945000e+03   
25%    1.344931e+06  1.346810e+06  1.340680e+06  1.436670e+06  1.483920e+06   
50%    8.840220e+06  9.043342e+06  9.148092e+06  9.219637e+06  9.298515e+06   
75%    5.478589e+07  5.657805e+07  5.843831e+07  5.882673e+07  5.909536e+07   
max    6.511725e+09  6.592712e+09  6.674182e+09  6.757000e+09  6.839554e+09   

       ...          2012          2013          2014          2015  \
count  ...  2.640000e+02  2.640000e+02  2.640000e+02  2.640000e+02   
mean   ...  2.874902e+08  2.912969e+08  2.951160e+08  2.989277e+08   
std    ...  9.017511e+08  9.129343e+08  9.241050e+08  9.352101e+08   
min    ...  1.013600e+04  1.020800e+04  1.028900e+04  1.037400e+04   
25%    ...  1.539939e+06  1.574621e+06  1.609909e+06  1.645868e+06   
50%    ...  9.824808e+06  9.948838e+06  1.001582e+07  1.022085e+07   
75%    ...  6.057984e+07  6.120753e+07  6.174243e+07  6.182699e+07   
max    ...  7.089255e+09  7.175500e+09  7.261847e+09  7.347679e+09   

               2016          2017          2018          2019          2020  \
count  2.640000e+02  2.640000e+02  2.640000e+02  2.640000e+02  2.640000e+02   
mean   3.027560e+08  3.065980e+08  3.103591e+08  3.140425e+08  3.176734e+08   
std    9.463321e+08  9.575052e+08  9.683483e+08  9.788967e+08  9.891628e+08   
min    1.047400e+04  1.057700e+04  1.067800e+04  1.076400e+04  1.083400e+04   
25%    1.689616e+06  1.716772e+06  1.740174e+06  1.751950e+06  1.767996e+06   
50%    1.036160e+07  1.040671e+07  1.045548e+07  1.047907e+07  1.052565e+07   
75%    6.187352e+07  6.191725e+07  6.193141e+07  6.150589e+07  6.157091e+07   
max    7.433651e+09  7.519371e+09  7.602716e+09  7.683806e+09  7.763933e+09   

               2021  
count  2.640000e+02  
mean   3.210893e+08  
std    9.988295e+08  
min    1.087300e+04  
25%    1.791783e+06  
50%    1.054019e+07  
75%    6.295547e+07  
max    7.836631e+09  

[8 rows x 22 columns]

# Questions

1. In the year 2007, provide the country names where the population is between 175000-58000000.
2. What is the standard deviation (std) for the population in 2019.
3. For the population in 2003, filter out the population that are in the 3rd quartile (75% and above) and list out the country names.

# SOLUTION

# 1. In the year 2007, provide the country names where the population is between 175000-58000000.

I want to print the first 7 rows of my data frame to get a closer view for analysis

In [3]:
df.head(7)

Country Name Country Code         2000         2001  \
0                        Aruba          ABW      90866.0      92892.0   
1  Africa Eastern and Southern          AFE  398113044.0  408522129.0   
2                  Afghanistan          AFG   20779957.0   21606992.0   
3   Africa Western and Central          AFW  267214544.0  274433894.0   
4                       Angola          AGO   16395477.0   16945753.0   
5                      Albania          ALB    3089027.0    3060173.0   
6                      Andorra          AND      65390.0      67344.0   

          2002         2003         2004         2005         2006  \
0      94992.0      97016.0      98744.0     100028.0     100830.0   
1  419223717.0  430246635.0  441630149.0  453404076.0  465581372.0   
2   22600774.0   23680871.0   24726689.0   25654274.0   26433058.0   
3  281842480.0  289469530.0  297353098.0  305520588.0  313985474.0   
4   17519418.0   18121477.0   18758138.0   19433604.0   20149905.0   
5    3051010.0    3039616.0    3026939.0    3011487.0    2992547.0   
6      70048.0      73180.0      76250.0      78871.0      80995.0   

          2007  ...         2012         2013         2014         2015  \
0     101226.0  ...     102565.0     103165.0     103776.0     104339.0   
1  478166911.0  ...  547482863.0  562601578.0  578075373.0  593871847.0   
2   27100542.0  ...   31161378.0   32269592.0   33370804.0   34413603.0   
3  322741656.0  ...  370243017.0  380437896.0  390882979.0  401586651.0   
4   20905360.0  ...   25107925.0   26015786.0   26941773.0   27884380.0   
5    2970017.0  ...    2900401.0    2895092.0    2889104.0    2880703.0   
6      82682.0  ...      82427.0      80770.0      79213.0      77993.0   

          2016         2017         2018         2019         2020  \
0     104865.0     105361.0     105846.0     106310.0     106766.0   
1  609978946.0  626392880.0  643090131.0  660046272.0  677243299.0   
2   35383028.0   36296111.0   37171922.0   38041757.0   38928341.0   
3  412551299.0  423769930.0  435229381.0  446911598.0  458803476.0   
4   28842482.0   29816769.0   30809787.0   31825299.0   32866268.0   
5    2876101.0    2873457.0    2866376.0    2854191.0    2837849.0   
6      77295.0      76997.0      77008.0      77146.0      77265.0   

          2021  
0     107195.0  
1  694665117.0  
2   39835428.0  
3  470898870.0  
4   33933611.0  
5    2811666.0  
6      77354.0  

[7 rows x 24 columns]

In [4]:
# I am only interested in 2 columns, the 'Country Name' and the '2007' column. I will print this 2 columns;

df1 = df[['Country Name','2007']]
df1

Country Name         2007
0                          Aruba     101226.0
1    Africa Eastern and Southern  478166911.0
2                    Afghanistan   27100542.0
3     Africa Western and Central  322741656.0
4                         Angola   20905360.0
..                           ...          ...
261                       Kosovo    1733404.0
262                  Yemen, Rep.   21282514.0
263                 South Africa   49119766.0
264                       Zambia   12502958.0
265                     Zimbabwe   12255920.0

[266 rows x 2 columns]

In [5]:
# I want to rename the 2007 column so it does not affect my calculation since it is a number. I will call the column Two_Seven.

df1 = df1.rename(columns={'2007':'Two_Seven'}) 
df1.head(7)

Country Name    Two_Seven
0                        Aruba     101226.0
1  Africa Eastern and Southern  478166911.0
2                  Afghanistan   27100542.0
3   Africa Western and Central  322741656.0
4                       Angola   20905360.0
5                      Albania    2970017.0
6                      Andorra      82682.0

In [6]:
# There are 266 rows here, i will compare the number of rows in my final answer with this.

# Now lets obtain the dataframe for country name and population in 2007.

new_df1 = df1[(df1.Two_Seven >= 175000) & (df1.Two_Seven <= 58000000)]
new_df1

Country Name   Two_Seven
2             Afghanistan  27100542.0
4                  Angola  20905360.0
5                 Albania   2970017.0
8    United Arab Emirates   6168846.0
9               Argentina  39684303.0
..                    ...         ...
261                Kosovo   1733404.0
262           Yemen, Rep.  21282514.0
263          South Africa  49119766.0
264                Zambia  12502958.0
265              Zimbabwe  12255920.0

[162 rows x 2 columns]

In [7]:
# To print only the 'Country Name', we use the 'drop' function to drop the 'Two_Seven' column

final_df1 = new_df1.drop(['Two_Seven'], axis=1)
final_df1

Country Name
2             Afghanistan
4                  Angola
5                 Albania
8    United Arab Emirates
9               Argentina
..                    ...
261                Kosovo
262           Yemen, Rep.
263          South Africa
264                Zambia
265              Zimbabwe

[162 rows x 1 columns]

We can see that the number of rows have reduced to 162 in our final answer as compared to 266 when we started.

# 2. What is the standard deviation (std) for the population in 2019.

from df.describe() carried out above, it is shown that the standard deviation(std) for 2019 population is 9.788967e+08
(+08 means times 10 raised to power of positive 8). The figure is equivalent to 978896700

We can also just use the df['column name'].std() function. But first we are only interested in the 2019 column.

In [8]:
df2 = df['2019'].std()
df2

978896713.8254997

In [9]:
# This gives the same result as before.

# 3. For the population in 2003, filter out the population that are in the 3rd quartile (75% and above) and list out the Country Name.

To answer this, we need only the 'Country Name' and the '2003' column. 

In [10]:
# I want to display only the 'Country Name' and '2003' column in the dataset

df3 = df[['Country Name','2003']]
df3

Country Name         2003
0                          Aruba      97016.0
1    Africa Eastern and Southern  430246635.0
2                    Afghanistan   23680871.0
3     Africa Western and Central  289469530.0
4                         Angola   18121477.0
..                           ...          ...
261                       Kosovo    1703466.0
262                  Yemen, Rep.   18985001.0
263                 South Africa   46719203.0
264                       Zambia   11256740.0
265                     Zimbabwe   11982219.0

[266 rows x 2 columns]

In [11]:
# From this, the df3 dataset has 266 rows. I will compare this after my final answer.

# The describe function will provide the third quartile (75% and above) for me, though I would preferably use another method to show it.

df3.describe()

2003
count  2.650000e+02
mean   2.545388e+08
std    8.060498e+08
min    9.724000e+03
25%    1.283564e+06
50%    8.557160e+06
75%    5.142558e+07
max    6.351856e+09

In [12]:
# we can see 75% (Third Quartile) is 5.142558e+07  (+07 means times 10 raised to power +7).

# I want to rename the 2003 column so it does not affect my calculation since it is a number. I will call the column Two_Three.

df3 = df3.rename(columns={'2003':'Two_Three'}) 
df3.head(7)

Country Name    Two_Three
0                        Aruba      97016.0
1  Africa Eastern and Southern  430246635.0
2                  Afghanistan   23680871.0
3   Africa Western and Central  289469530.0
4                       Angola   18121477.0
5                      Albania    3039616.0
6                      Andorra      73180.0

In [13]:
# To obtain population in 2003 that are above 75%, we need to obtain values in the third Quartile (Q3)

Q3 = df3.Two_Three.quantile(0.75)
Q3

51425583.0

In [14]:
# Notice that the value of Q3 is the same as the 75% value when i did df3.describe().

# Now that i have a base value, Q3 which accounts for 75% of the value in 2003, I will just display the country names that are above this value.

# for convenience i will call the new dataframe dfQ3.

In [15]:
dfQ3 = df3[(df3.Two_Three > Q3)]
dfQ3

Country Name     Two_Three
1    Africa Eastern and Southern  4.302466e+08
3     Africa Western and Central  2.894695e+08
7                     Arab World  3.011139e+08
20                    Bangladesh  1.347916e+08
29                        Brazil  1.818092e+08
..                           ...           ...
244                      Turkiye  6.608940e+07
249          Upper middle income  2.206018e+09
251                United States  2.901079e+08
257                      Vietnam  8.230165e+07
259                        World  6.351856e+09

[66 rows x 2 columns]

In [16]:
# Now, i would create a new dataframe to drop the 'Two_three' column since it is not needed for my result.
# I will call the new dataframe dfQ3.

new_dfQ3 = dfQ3.drop(['Two_Three'], axis = 1)
new_dfQ3

Country Name
1    Africa Eastern and Southern
3     Africa Western and Central
7                     Arab World
20                    Bangladesh
29                        Brazil
..                           ...
244                      Turkiye
249          Upper middle income
251                United States
257                      Vietnam
259                        World

[66 rows x 1 columns]

In [17]:
# Observe that in the final answer, the row has reduced from 266 rows in df3 to 66 rows in new_dfQ3